onecgiar-pr-server/src/api/results/share-result-request/share-result-request.service.ts

```diff
@@ -121,6 +481,321 @@ export class ShareResultRequestService {
+        }
+      }
+    }
+
+    return {
+      pendingOwner:
+        role !== 1
+          ? {
+              ...commonConditions,
+              shared_inititiative_id: In(sharedInitiativeIds),
+              is_map_to_toc: false,
+            }
+          : commonConditions,
+      pendingShared:
+        role !== 1
+          ? {
+              ...commonConditions,
+              owner_initiative_id: In(sharedInitiativeIds),
+              is_map_to_toc: true,
+            }
+          : commonConditions,
+      done: [
+        {
+          ...commonConditions,
+          request_status_id: In([2, 3]),
+          is_active: true,
+          obj_result: { is_active: true },
+          shared_inititiative_id: In(sharedInitiativeIds),
+          is_map_to_toc: false,
+        },
+        {
+          ...commonConditions,
+          request_status_id: In([2, 3]),
+          is_active: true,
+          obj_result: { is_active: true },
+          owner_initiative_id: In(sharedInitiativeIds),
+          is_map_to_toc: true,
+        },
+      ],
+    };
+  }
+
+  private async getRequest(whereCondition: any) {
+    return await this._shareResultRequestRepository.find({
+      select: this.getRequestSelectFields(),
+      relations: this.getRequestRelations(),
+      where: whereCondition,
+    });
+  }
+
+  private getRequestSelectFields() {
+    return {
+      share_result_request_id: true,
+      result_id: true,
+      requested_date: true,
+      aprovaed_date: true,
+      request_status_id: true,
+      is_map_to_toc: true,
+      obj_request_status: {
+        request_status_id: true,
+        name: true,
+      },
+      obj_result: {
+        result_code: true,
+        title: true,
+        status_id: true,
+        obj_version: {
+          id: true,
+          status: true,
+          phase_name: true,
+        },
+        obj_result_type: {
+          id: true,
+          name: true,
+        },
+        obj_result_level: {
+          id: true,
+          name: true,
+        },
+        obj_results_toc_result: {
+          result_toc_result_id: true,
+          initiative_id: true,
+          is_active: true,
+        },
+      },
+      obj_requested_by: {
+        id: true,
+        first_name: true,
+        last_name: true,
+      },
+      obj_approved_by: {
+        id: true,
+        first_name: true,
+        last_name: true,
+      },
+      obj_owner_initiative: {
+        id: true,
+        official_code: true,
+        name: true,
+      },
+      obj_shared_inititiative: {
+        id: true,
+        official_code: true,
+        name: true,
+      },
+    };
+  }
+
+  private getRequestRelations() {
+    return {
+      obj_request_status: true,
+      obj_result: {
+        obj_version: true,
+        obj_result_type: true,
+        obj_result_level: true,
+      },
+      obj_requested_by: true,
+      obj_approved_by: true,
+      obj_owner_initiative: true,
+      obj_shared_inititiative: true,
+    };
+  }
```

Comment:
@JuanGuzman-io please consider typing this as `FindOptionsRelations<you-entity>`